pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -237,20 +300,54 @@ func (c *Controller) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 			Fn:           flow.TaskFn(botanist.WaitUntilControlPlaneReady).DoIf(cleanupShootResources && controlPlaneDeploymentNeeded && !shootNamespaceInDeletion),
 			Dependencies: flow.NewTaskIDs(deployControlPlane),
 		})
-		wakeUpControlPlane = g.Add(flow.Task{
-			Name:         "Waking up control plane to ensure proper cleanup of resources",
-			Fn:           flow.TaskFn(botanist.WakeUpControlPlane).DoIf(wakeupRequired),
-			Dependencies: flow.NewTaskIDs(waitUntilControlPlaneReady, ensureShootClusterIdentity),
+		generateEncryptionConfigurationMetaData = g.Add(flow.Task{
+			Name:         "Generating etcd encryption configuration",
+			Fn:           flow.TaskFn(botanist.GenerateEncryptionConfiguration).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(enableEtcdEncryption && cleanupShootResources),
+			Dependencies: flow.NewTaskIDs(ensureShootStateExists),
+		})
+		createOrUpdateETCDEncryptionConfiguration = g.Add(flow.Task{
+			Name:         "Applying etcd encryption configuration",
+			Fn:           flow.TaskFn(botanist.ApplyEncryptionConfiguration).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(enableEtcdEncryption && cleanupShootResources),
+			Dependencies: flow.NewTaskIDs(ensureShootStateExists, generateEncryptionConfigurationMetaData),
+		})
+		deployKubeAPIServer = g.Add(flow.Task{
+			Name: "Deploying Kubernetes API server",
+			Fn:   flow.TaskFn(botanist.DeployKubeAPIServer).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(cleanupShootResources),
+			Dependencies: flow.NewTaskIDs(
+				deploySecrets,
+				deployETCD,
+				waitUntilEtcdReady,
+				waitUntilKubeAPIServerServiceIsReady,
+				waitUntilControlPlaneReady,
+				createOrUpdateETCDEncryptionConfiguration,
+			).InsertIf(!staticNodesCIDR),
+		})
+		_ = g.Add(flow.Task{
+			Name: "Scale up Kubernetes API server",
+			Fn: flow.TaskFn(botanist.ScaleKubeAPIServerToOne).
+				RetryUntilTimeout(defaultInterval, defaultTimeout).
+				DoIf(cleanupShootResources && kubeAPIServerDeploymentReplicas == 0),
```

Comment:
cc @dguendisch  - I guess this also fixes the problem that we sporadically see in the integration tests (KAPI scaled down to 0)